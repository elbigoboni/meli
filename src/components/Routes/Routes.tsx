import React from "react";
import { Route } from "react-router-dom";
import { Detail } from "../../pages/Detail";
import { Details } from "../../pages/Details";
import { Home } from "../../pages/Home";
import { Header } from "../Header";

const Routes: React.FC = () => (
  <>
    <Header />
    <Route path="/" exact={true}>
      <Home />
    </Route>
    <Route path="/items/:s" exact={true}>
      <Details />
    </Route>
    <Route path="/item/:id" exact={true}>
      <Detail />
    </Route>
  </>
);

export { Routes };
