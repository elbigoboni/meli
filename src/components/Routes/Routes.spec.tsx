import { render } from "@testing-library/react";
import React from "react";
import { MemoryRouter } from "react-router-dom";
import { Routes } from "./Routes";

describe("<Routes />", () => {
  describe("Render", () => {
    test("should render correctly", () => {
      const { container } = render(<Routes />, { wrapper: MemoryRouter });
      expect(container).toMatchSnapshot();
    });
  });
});
