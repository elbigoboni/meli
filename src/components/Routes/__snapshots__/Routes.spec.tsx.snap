// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Routes /> Render should render correctly 1`] = `
<div>
  <header
    class="sc-iJuUWI kNRSja"
  >
    <header
      class="sc-iqHYGH fCINnt"
    >
      <div
        class="sc-ezrdKe kShTdh"
      >
        <a
          data-testid="link to home"
          href="/"
        >
          <img
            class="sc-giIncl fhSfPO"
            data-testid="logo"
            src="/images/Logo_ML@2x.png"
          />
        </a>
        <form
          class="sc-bBXqnf joqZBh"
          data-testid="search box"
        >
          <div
            class="sc-lmoMRL kzUTia"
          >
            <input
              class="sc-iwyYcG jCDFlQ"
              data-testid="search field"
              placeholder="Nunca dejes de buscar"
              type="text"
              value=""
            />
            <button
              class="sc-cxFLnm cqWXAG"
              type="submit"
            >
              Pesquisar
            </button>
          </div>
        </form>
      </div>
    </header>
  </header>
  <header
    class="sc-iqHYGH fCINnt"
  >
    <img
      class="sc-fKFyDc dqlKDa"
      src="/images/shopping.svg"
    />
  </header>
</div>
`;
